CV Backend/src/main/java/com/qa/demo/Interoperability/AccountController.java

```diff
@@ -0,0 +1,233 @@
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.Principal;
+import java.util.Base64;
+import java.util.List;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+
+@Transactional
+@RestController
+@RequestMapping("/account")
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class AccountController {
+
+    @Autowired
+    private AccountRepository repository;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+    
+    @Autowired
+    private PasswordEncoder bCode;
+
+    @Autowired
+    private MongoUserDetailsService mongoUserDetailsService;
+
+    @PostAuthorize("hasRole('ROLE_admin')")
+    @GetMapping(value = "/getall")
+    public List<Account> getAllAccounts(){
+        return repository.findAll();
+    }
+
+    @PostMapping(value = "/")
+    public Account createAccount(@Valid @RequestBody Account account){
+        System.out.println("Account creation attempting");
+        account.set_id(ObjectId.get());
+        //account.setUserRole("user");
+        //System.out.println(bCode.encode(account.getPassword()));
+        account.setPassword(bCode.encode(account.getPassword()));
+        repository.save(account);
+        return account;
+    }
+
+//    @DeleteMapping(value = "/delete/{id}")
+//    public String deleteFileFromUser(@PathVariable("id") ObjectId id){
+//
+//    }
+
+    @PostMapping(value = "/upload/{id}")
+    public String addFileToUser(@RequestParam("file") MultipartFile multipartFile, @PathVariable("id") ObjectId id){
+        System.out.println("File uploading being attempted");
+        Account account = repository.findBy_id(id);
+        try {
+            mongoTemplate.upsert(Query.query(where("_id").is(id)), Update.update("file",new Binary(BsonBinarySubType.BINARY, multipartFile.getBytes())), Account.class);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "File addition failed";
+        }
+        return "File addition successful to account:"+account.getFirstName()+" "+account.getLastName();
+    }
+
+    @GetMapping(value = "/download/{id}")
+    public ResponseEntity<byte[]> getUserFile(@PathVariable("id") ObjectId id){
+        System.out.println("File download being attempted");
+        Account account = repository.findBy_id(id);
+        Binary accountCV = repository.findBy_id(id).getFile();
+        FileOutputStream fileOutputStream = null;
+        if(accountCV != null){
```

Comment:
once again you're doing the findBy_id(id) but since you're not guaranteed the result you do if not null chech, initialise the variables and then do this findBy_id(id) inside the try block

Reply:
done